packages/zoe/src/zoe.js

```diff
@@ -428,134 +320,67 @@ function makeZoe(additionalEndowments = {}, vatPowers = {}) {
-        // We may want to handle this with static checking instead.
-        // Discussion at: https://github.com/Agoric/agoric-sdk/issues/1017
-        assert(
-          offerHandles.length >= 2,
-          details`reallocating must be done over two or more offers`,
-        );
-        assert(
-          offerHandles.length === newAllocations.length,
-          details`There must be as many offerHandles as entries in newAllocations`,
-        );
-
-        // 1) Ensure 'offer safety' for each offer separately.
-        const makeOfferSafeReallocation = (offerHandle, newAllocation) => {
-          const { proposal, currentAllocation } = offerTable.get(offerHandle);
-          const reallocation = harden({
-            ...currentAllocation,
-            ...newAllocation,
-          });
-
-          assert(
-            isOfferSafe(getAmountMathForBrand, proposal, reallocation),
-            details`The reallocation was not offer safe`,
-          );
-          return reallocation;
-        };
-
-        // Make the reallocation and test for offer safety by comparing the
-        // reallocation to the original proposal.
-        const reallocations = offerHandles.map((offerHandle, i) =>
-          makeOfferSafeReallocation(offerHandle, newAllocations[i]),
-        );
-
-        // 2. Ensure that rights are conserved overall.
-        const flattened = arr => [].concat(...arr);
-        const flattenAllocations = allocations =>
-          flattened(allocations.map(allocation => Object.values(allocation)));
-
-        const currentAllocations = offerTable
-          .getOffers(offerHandles)
-          .map(({ currentAllocation }) => currentAllocation);
-        const previousAmounts = flattenAllocations(currentAllocations);
-        const newAmounts = flattenAllocations(reallocations);
-
-        assert(
-          areRightsConserved(
-            getAmountMathForBrand,
-            previousAmounts,
-            newAmounts,
-          ),
-          details`Rights are not conserved in the proposed reallocation`,
-        );
-
-        // 3. Save the reallocations.
-        offerTable.updateAmounts(offerHandles, reallocations);
-      },
-
-      complete: offerHandles => {
-        assert(
-          Array.isArray(offerHandles),
-          details`zcf.complete(offerHandles) must be Array, not ${offerHandles}`,
-        );
-        assertOffersHaveInstanceHandle(offerHandles, instanceHandle);
-        return completeOffers(instanceHandle, offerHandles);
-      },
+  // Make a Zoe invite payment with an value that is a mix of credible
+  // information from Zoe (the `handle` and `instanceHandle`) and
+  // other information defined by the smart contract (the mandatory
+  // `inviteDesc` and the optional `options.customProperties`).
+  // Note that the smart contract cannot override or change the values
+  // of `handle` and `instanceHandle`.
+  const makeInvitation = (
+    instanceHandle,
+    offerHook,
```

Comment:
This is no longer an offerHook function, right?
```suggestion
    { offerHandler },
```
What if we destructure immediately?

Reply:
zcf calls `E(zoeForZcf).makeInvitation(inviteCallback, inviteDesc, options)`.

zoeForZcf defines it (line 372) as `makeInvitation: (...params) => makeInvitation(instanceHandle, ...params)`. 

so `instanceHandle` is added before the params sent by zcf. Zcf's first argument was a callback, which this method stores in inviteHandleToCallback.  I'll rename it to callback, since that's how it's used.